src/mdformat/_cli.py

```diff
@@ -303,6 +303,17 @@ def make_arg_parser(
             plugin.add_cli_argument_group(group)
             for action in group._group_actions:
                 action.dest = f"plugin.{plugin_id}.{action.dest}"
+                if action.default not in {None, argparse.SUPPRESS}:
+                    import warnings
+
+                    text = f"The argument default for {action.option_strings} from the '{plugin_id}' plugin, will always override any value configured in TOML. The only supported CLI defaults are `None` or `argparse.SUPPRESS`"  # noqa: E501
```

Comment:
Feel free to revert the last commit before merging if you prefer the string to be hard-wrapped.

I thought black would manage strings like that automatically, but I'm not sure if there a missing config in pre-commit or a version issue?

Reply:
Black doesn't wrap strings, to my knowledge (I could be wrong). I don't mind the wrapping, either way works for me.

I really liked how you had concrete instructions on how to fix, in the previous edit. Could you add something similar like ```To resolve, do `.add_argument(..., default=None)` ``` ?

Maybe also move the string inside `warn_explicit` to not pollute the namespace with a generic name.

Reply:
I've updated the description above (https://github.com/hukkin/mdformat/pull/512#issue-2825317428) with the new output. Feel free to edit the text directly or let me know if you have other feedback!